2015/proposals/girdhar-yashasvi.md

```diff
@@ -0,0 +1,253 @@
+
+
+##Why this project.
+
+Firstly, the reason of choosing Software Carpentry for my summer project is that I strongly believe in the
+vision of the organisation and wanted a way to get involved with it. I think being a Google Summer of Code intern
+this year is one of the best ways to accomplish that.
+It would be a great learning experience for me to observe how the organisation works and I wish to remain
+connected with it even after my project.
+
+Now, the reason for choosing this project is that I am interested in implmenting what is required in the project
+and I think I have the required experience for it. This project attracts me because it provides me a way to use my
+skill set at an organisational level and if completed successfully, it would be a very helpful tool for Software
+Carpentry in future.
+
+I am confident that if I am offered this opportunity, I will be able to contribute positively. I wish to be in a
+long and creative relationship with Software Carpentry and this project would just be a start to it.
+
+<a name="schema"></a>
+###Schema :
+
+I have assumed that the information related to a particular workshop may also be required.
+If that’s not the case, I have designed the schema in such a way that it can easily be changed.
+
+
+####Table 1: Result Entries
+
+| id  | system_info  | upload_timing | workshop_id |
+| --- | ------------ | ------------- | ----------- |
+
+####Table 2: Package Requirements
+
+| workshop_id | package_name | required_version |
+| ----------- | ------------ | ---------------- |
+
+workshop_id + package_name is the primary key here
+
+####Table 3: Package Test Entries
+
+| uid  | pname  | installed_version | diagnostic_information |
+| ---- | ------ | ----------------- | ---------------------- |
+
+uid-> foreign key to primary key(id) of Result Entries
+
+
+Here are some types of sample queries that the system can handle.
+
+1. What fraction of last month’s submitting hosts had Git < 1.8.
+
+2. How many tests failed in a particular workshop.
+
+3. How many tests failed in a particular workshop on Windows.
+
+4. What fraction of failed tests were on OS X for a specific time period.
+
+5. For a particular workshop, find number of failed version test for Git.
+
+6. For a particular workshop, information about all the packages.
+
+
+7. Total failed/passed checks for the past 10 workshops OR for the workshops held between a specific time period.
+
+
+8. Similarly, total failed version tests on pytest package in the last 10 months.
+
+
+9. Find the required versions of all packages of a specific workshop id.
+
+Some concepts used:
+
+* user entries and packages are related as many to many, there I need to make a third table(Table 3) for them which contains two foreign keys, pointing to the primary keys of user entries and packages.
+
+
+So basically, from what I understand, in any query (that asks for failed/passed tests or version tests) we should be able to add :
```

Comment:
What do you think you don't understand?  If you don't have anything in particular that's bugging you, I'd just use “For any query we should be able to add:”.  “From what I understand” is generally implied in anything serious that someone says ;).